trending_data/src/main/webapp/unique_window_handler.js

```diff
@@ -0,0 +1,191 @@
+    this.twitterDataDiv = document.createElement('div');
+  }
+
+  /**
+  * DataWindow contains a button container with the 2
+  * buttons (YouTube, Twitter) and the content corresponding
+  * to each platform. This method initializes the DataWindow
+  * with the button container.
+  */
+  initDataWindow() {
+    this.dataWindow = document.createElement('div');
+    this.dataWindow.className = 'popup';
+    this.dataWindow.appendChild(this.createBttnDiv());
+  }
+
+  /**
+  * Called when Twitter button is clicked. Clears
+  * current content of window and shows the most recently
+  * cached Twitter data
+  */
+  showYTData() {
+    this.initDataWindow(); // clear current content and re-add buttons
+    this.dataWindow.appendChild(this.ytDataDiv);
+    this.currentWindow.setContent(this.dataWindow);
+  }
+
+  /**
+  * Called when YouTube button is clicked. Clears
+  * current content of window and shows the most recently
+  * cached YouTube data
+  */
+  showTwitterData() {
+    this.initDataWindow();
+    this.dataWindow.appendChild(this.twitterDataDiv);
+    this.currentWindow.setContent(this.dataWindow);
+  }
+
+  /**
+  * Creates a container with 2 buttons, one for Twitter
+  * and one for YouTube and adds them to the top of the
+  * window where data will be displayed
+  * @return {HTMLElement} bttnDiv
+  */
+  createBttnDiv() {
+    const bttnDiv = document.createElement('div');
+    bttnDiv.className = 'toggle-btns';
+
+    /** YouTube button */
+    const ytBttn = document.createElement('button');
+    ytBttn.textContent = 'YouTube';
+    ytBttn.className = 'btn btn-danger';
+    bttnDiv.appendChild(ytBttn);
+
+    /** Twitter button */
+    const twitterBttn = document.createElement('button');
+    twitterBttn.textContent = 'Twitter';
+    twitterBttn.className = 'btn btn-info';
+    bttnDiv.appendChild(twitterBttn);
+
+    /** Toggle platforms. If YouTube button is clicked ->
+      show YouTube data. If Twitter button is clicked ->
+      show Twitter data */
+    ytBttn.addEventListener('click', () => {
+      this.showYTData();
+    });
+
+    twitterBttn.addEventListener('click', () => {
+      this.showTwitterData();
+    });
+
+    return bttnDiv;
+  }
+
+  /**
+  * Checks if the current window is open
+  * @return {boolean}
+  */
+  isInfoWindowOpen() {
+    if (this.currentWindow == null) {
+      return false;
+    } else {
+      const map = this.currentWindow.getMap();
+      return (map !== null && typeof map !== 'undefined');
+    }
+  }
+
+  /**
+  * Checks if the current window is open
+  * @param {Marker} marker
+  * @return {boolean}
+  */
+  isCkickOnDifferentWindow(marker) {
+    return (!this.isInfoWindowOpen() ||
+      marker.countryCode != this.getCountryCode());
+  }
+
+  /**
+  * Loads the current YouTube data and opens a window
+  * which contains it
+  * @param {Marker} marker
+  * @param {HTMLElement} content
+  */
+  async openWindow(marker) {
+    const content = await this.loadYTData(marker);
```

Comment:
maybe change this to `ytContent` now that the method is no longer called `loadYTDataAndOpenWindow` and it's not clear what that content represents